apps/prairielearn/src/ee/auth/azure/callback.ts

```diff
@@ -1,13 +1,12 @@
-// @ts-check
-const passport = require('passport');
-const express = require('express');
-const asyncHandler = require('express-async-handler');
+import * as passport from 'passport';
+import { Router } from 'express';
+import asyncHandler = require('express-async-handler');
 
-const authnLib = require('../../../lib/authn');
+import * as authnLib from '../../../lib/authn';
 
-const router = express.Router();
+const router = Router();
 
-function authenticate(req, res) {
+function authenticate(req, res): Promise<any> {
```

Comment:
We can't import `@types/passport` or something to get a better return type here?

Reply:
We already have passport types; passport just has no way of tying the first argument (`'azuread-openidconnect'`) to a particular shape for a user object. The best we could do is to define that type ourselves and annotate this function with it (or maybe validate explicitly with zod).